modules/internal/LocaleTree.chpl

```diff
@@ -51,4 +51,51 @@ module LocaleTree {
   }
 
   chpl_initLocaleTree();
+
+  iter localeTree(root = here, parent = here) : locale {
+    halt("DONT USE SERIAL ITER");
+    yield parent;
+  }
+
+  /*
+   Logarithmic gather iterator
+   */
+  iter localeTree(param tag:iterKind, root = here, parent = here) : locale where tag == iterKind.standalone {
```

Comment:
>I would want something called localeTree to actually do an on-statement onto each locale.

Fair enough. I wasn't particularly attached to the current naming scheme, and was hoping to get some feedback on that. Have any ideas? Maybe `localeGather` and `localeBroadcast`?